src/WaxSigningApi.ts

```diff
@@ -164,9 +191,11 @@ export class WaxSigningApi {
     noModify = false,
     feeFallback = true
   ): Promise<ISigningResponse> {
+    const startTime = Math.floor(new Date().getTime());
     const confirmationWindow: Window = await this.waxEventSource.openEventSource(
       `${this.waxSigningURL}/cloud-wallet/signing/`,
       {
+        startTime,
```

Comment:
this is unnecesary. We don't have to pass this around through the back end. You have to use javascript variable binding to associate the start time to your requests

Reply:
if you change the parameters of receiveSignatures like I suggest, you can then in line 210 replace `this.receiveSignatures.bind(this)` with `this.receiveSignatures.bind(this, startTime, "login.metric.manual_sign_transaction_time")`

Reply:
you will also have to do the same in line 185 which also has the `this.receiveSignatures({ data });", but you should give it a different and better metrics name to distinguish it as signing via endpoint vs signing vua manual signing